extraction/extractor.py

```diff
@@ -0,0 +1,91 @@
+"""This script is used to scrape data from URL, extract data from the DOM
+    and store the extracted data."""
+
+import re
+
+import requests
+from bs4 import BeautifulSoup, Comment
+from data_models import contents, text  # pylint: disable-msg=E0401
+
+REQUEST_SESSION = requests.Session()
+pattern_for_video_ext = re.compile(r'^.*\.(mp4|ogg|mov|webm)')
+pattern_for_ads = re.compile('(^ad-|^ads-)')
+extra_tags = ['script', 'noscript', 'style', 'header',
+              'footer', 'meta', 'link']
+
+
+class Extractor:
+    """This class takes the URL and extracts the data(DOM) from it using
+    Beautiful Soup and stores the extracted data in contents_list"""
+
+    def __init__(self, url):
+        self.url = url
+        self.soup = None
+        self.contents_list = contents.Contents()
+
+    def extract_html(self):
+        """This function parses data from HTML using Beautiful Soup"""
+
+        file = REQUEST_SESSION.get(self.url).text    # To request Html from URL
+
+        # Remove the comment to read local files for testing purpose
+        # file = open('./test_html.html','r').read()
+
+        self.soup = BeautifulSoup(file, 'lxml')
+        self.clean_soup()
+
+        self.extract_data_from_html()
+
+    def clean_soup(self):
```

Comment:
Can you prepend methods which are meant to private with '_' 
For this class extract_html is the public method this class exposes, the others are internal implementation details right

Reply:
Updated accordingly!